example_code/amo_scenario_3.c

```diff
@@ -1,16 +1,21 @@
 #include <shmem.h>
 
 int main(void) {
-    static int x = 0;
+    static long psync[SHMEM_REDUCE_SYNC_SIZE];
+    static int pwrk[SHMEM_REDUCE_MIN_WRKDATA_SIZE];
+    static int x = 0, y = 0;
+
+    for (int i = 0; i < SHMEM_REDUCE_SYNC_SIZE; i++)
+        psync[i] = SHMEM_SYNC_VALUE;
 
     shmem_init();
-    /* Undefined behavior: OpenSHMEM atomic increment operations are concurrent with the local
-     * increment of symmetric variable 'x'. */
-    if (shmem_my_pe() > 0)
-      shmem_int_atomic_inc(&x, 0);
-    else
-      x++;
+    shmem_int_atomic_inc(&x, (shmem_my_pe()+1) % shmem_n_pes());
+    /* Undefined behavior: The following reduction operation performs accesses to symmetric
+     * variable 'x' that are concurrent with previously issued atomic increment operations
+     * on the same variable. */
+    shmem_int_sum_to_all(&y, &x, 1, 0, 0, shmem_n_pes(), pwrk, psync);
```

Comment:
Do we want to use the team-based collective API?

Reply:
We could&mdash;and maybe should&mdash;make this change, but I didn't actually change this example in this PR. It's hard to see in the GitHub diff, but I structured the commits to help show that I renamed the examples to put the atomicity domain example first. So, I didn't actually change this example, though the diff on the PR shows otherwise.